ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-sdk-e2e-tests.robot

```diff
@@ -35,28 +42,33 @@ Prepare Codeflare-sdk E2E Test Suite
     END
     Enable Component    codeflare
     Enable Component    ray
-    Create Directory    %{WORKSPACE}/codeflare-sdk-e2e-logs
+    ${result} =    Run Process  ./ods_ci/tests/Resources/Page/DistributedWorkloads/setup_python_virtual_env.sh
+    ...    shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+    IF    ${result.rc} != 0
+        FAIL    Unable to setup Python
+    END
     RHOSi Setup
 
 Teardown Codeflare-sdk E2E Test Suite
     [Documentation]    Teardown codeflare-sdk E2E Test Suite
     Disable Component    codeflare
     Disable Component    ray
-    RHOSi Teardown
+    ${result} =    Run Process  ./ods_ci/tests/Resources/Page/DistributedWorkloads/cleanup_python_virtual_env.sh
+    ...    shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+    IF    ${result.rc} != 0
+        FAIL   Unable to cleanup Python virtual environment
+    END
 
 Run Codeflare-sdk E2E Test
     [Documentation]    Run codeflare-sdk E2E Test
-    [Arguments]    ${test_name}
-    Log To Console    Running codeflare-sdk test: ${test_name}
-    ${result} =    Run Process    go test -timeout 30m -v ./tests/e2e -run ${test_name}
+    [Arguments]    ${TEST_NAME}
+    Log To Console    "Running codeflare-sdk test: ${TEST_NAME}"
+    ${result} =    Run Process  source venv3.9/bin/activate && cd codeflare-sdk && poetry env use 3.9 && poetry install --with test,docs && poetry run pytest -v -s ./tests/e2e/${TEST_NAME}
```

Comment:
IMHO it would have more sense to install latest released SDK version.
Current implementation test latest main branch, which won't be used by users. Users will use latest release (or version which is bundled in Notebook images).

Reply:
Yes. May be fetch the codeflare sdk version from[ notebooks pip file ](https://github.com/opendatahub-io/notebooks/blob/cdc08c6e846c02404fa119cab5cfe71eecba5ebf/jupyter/datascience/ubi9-python-3.9/Pipfile#L19) , clone and test against that released tag would make sense

Reply:
Done